docs/tech_concepts_and_architecture_overview.md

```diff
@@ -1,41 +1,44 @@
+2. Provide easy access to the graph through RESTful API's and GraphQL end-points.
+3. Allow the creation of cross Weaviate graphs over a P2P network.
+4. Allow fuzzy traversing over multiple-graphs in a P2P network.
+5. Uses a word vector based solution (_Contextionary_) to give linguistical context to words in the database.
+
+_Note: for the implementation-roadmap see the readme.md file_
 
 ## Architecture overview
-The Weaviate server offers an API for users, and uses a (graph) database to store the data.
 
-The rest of this section we describe the data model, the role of the contextionary, and the offered API's.
+The Weaviate server offers an API for users and uses a (graph) database to store the data.
+
+A user can consume the Weaviate service through a set of RESTful end-points or through the GraphQL API.
+
+Below you'll find an overview of the data model, the role of the contextionary, and the offered API's.
 
 ### Data model: Things, Actions, Keys
+
 - Weaviate stores three types of data: Things, Actions and Keys.
 - Keys are used both for authentication and authorization in Weaviate.
 - Owners of a key can create more keys; the new key points to the parent key that is used to create the key.
 - Permissions (read, write, delete, execute) are linked to a key.
 - Each piece of data is an instance of a Class. Each class is either a Thing or an Action.
 - A Class has a name, optional keywords, and properties.
-- A Class can have (primitive) properties (e.g. numbers, strings) and references to other Class instances.
-- Each piece of data (e.g. Things & Actions) is associated with a Key.
-- A Thing describes a (physical) object, an Action describes a change.
+- A Class can have (primitive) properties (e.g., numbers, strings) and references to other Class instances.
+- Each piece of data (e.g., Things & Actions) is associated with a Key.
+- A Thing describes a (physical) object; an Action describes a change.
 
 ### Contextionary
-The contextionary is an abstraction that allows us to easily work with trained Word Vectors.
-It consists of a Approximate k-Nearest Neighbours memory mapped data structure and a memory mapped
-index for words. This allows us to quickly find the Vector of a word. From a vector, we can find
-words that are close.
 
-This will be used in the next milestone to query remote data sources. In this milestone,
-it is used to ensure that the names of thing and action classes and their properties can be
-encoded to a vector.
+The contextionary is an abstraction that allows us to work with trained Word Vectors easily. It consists of an Approximate k-Nearest Neighbours memory mapped data structure and a memory mapped index for words. This allows us to find the Vector of a word quickly. From a vector, we can find words that are close.
+
+The contextionary will be used to query remote data sources. It is used to ensure that the names of thing and action classes and their properties can be encoded to a vector.
```

Comment:
Here we are talking about things in the future, whilst in the introduction we are talking about "the vision". This is confusing.